src/modules/commons/components/searchableSelect/SearchableSelect.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * @module modules/commons/components/searchableSelect/SearchableSelect
+ */
+
+/* TODO:
+ - Select Option with arrow key
+ - Enter should behave like a select click
+ - Esc should behave like a cancel action
+ - Rework selected update (and datatype ???)
+*/
+
+import { html } from 'lit-html';
+import { MvuElement } from '../../../MvuElement';
+import css from './searchableSelect.css';
+import { isNumber } from '../../../../utils/checks';
+
+const Update_Placeholder = 'update_placeholder';
+const Update_Options = 'update_options';
+const Update_Selected = 'update_selected';
+const Update_Search = 'update_search';
+
+/**
+ * General purpose implementation of a select-like component with integrated filtering.
+ *
+ * @property {string} placeholder='' - The placeholder to show when the search field is empty.
+ * @property {string|null} selected=null - The currently selected option.
+ * @property {string} search='' - The search term to filter through the provided options
+ * @property {Array<String>} options - Unfiltered options the user can choose from
+ * @property {function(changedState)} onChange - The Change callback function when the search input changes or an option is selected
+ * @fires onChange Fires when the search input changes or an option is selected
+ *
+ * @class
+ * @author herrmutig
+ */
+export class SearchableSelect extends MvuElement {
+	#hasPointer = false;
+	#availableOptions = [];
+
+	// eslint-disable-next-line no-unused-vars
+	#onChange = (changedState) => {};
+
+	constructor() {
+		super({
+			placeholder: 'Search...',
+			selected: null,
+			search: '',
+			options: []
+		});
+	}
+
+	onInitialize() {
+		document.addEventListener('click', () => {
```

Comment:
The event listener should be removed in `onDisconnect`.
See `Modal.js` for an example.